### PR TITLE
Drop the dev dependency from the TxPool

### PR DESCRIPTION
**Pull request by zivkovicmilos**
# Description

Fixes EDGE-61

This PR removes the dev dependency from the txpool, and fixes tests that are unsigned.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [x] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

**Comment by zivkovicmilos** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by zivkovicmilos** [comment]
recheck

File: txpool/txpool_test.go [hunk 1]
```diff
@@ -98,60 +93,79 @@ type result struct {
...
-		tx.Gas = invalidGasLimit
+		tx := newTx(defaultAddr, 0, 1)
+		tx.Value = big.NewInt(-5)
+
+		tx = signTx(tx)
 
 		assert.ErrorIs(t,
 			pool.addTx(local, tx),
-			ErrBlockLimitExceeded,
+			ErrNegativeValue,
```
Comment (dbrajovic) [inline txpool/txpool_test.go]: I would place the `signTx` as argument to `addTx`.

Like so (applies to all below cases):
`pool.addTx(local, signTx(tx))`

**Review by dbrajovic — APPROVED** [review]
LGTM

File: txpool/txpool_test.go (hunk 1 above)
Comment (zivkovicmilos) [reply txpool/txpool_test.go]: Inlined 👍 

652b947

File: e2e/framework/testserver.go [hunk 2]
```diff
@@ -490,12 +480,22 @@ func (t *TestServer) WaitForReady(ctx context.Context) error {
 	return err
 }
 
-func (t *TestServer) TxnTo(ctx context.Context, address web3.Address, method string) *web3.Receipt {
+func (t *TestServer) InvokeMethod(
+	ctx context.Context,
+	contractAddress types.Address,
+	method string,
+	fromAddress types.Address,
+	fromKey *ecdsa.PrivateKey,
```
Comment (Kourin1996) [inline e2e/framework/testserver.go:498]: It is possible to calculate `fromAddress` from `fromKey`?

File: e2e/framework/testserver.go [hunk 3]
```diff
@@ -351,17 +352,32 @@ func (t *TestServer) Start(ctx context.Context) error {
...
+	return t.Config.Signer.SignTx(transaction, privateKey)
+}
+
 // DeployContract deploys a contract with account 0 and returns the address
-func (t *TestServer) DeployContract(ctx context.Context, binary string) (web3.Address, error) {
+func (t *TestServer) DeployContract(
+	ctx context.Context,
+	binary string,
+	privateKey *ecdsa.PrivateKey,
+	sender types.Address,
```
Comment (Kourin1996) [inline e2e/framework/testserver.go]: Is is possible to calculate `sender` from `privateKey`?

**Review by Kourin1996 — APPROVED** [review]
Looks good to me. I've left minor comments. Please fix them if necessary

File: e2e/framework/testserver.go (hunk 2 above)
Comment (zivkovicmilos) [reply e2e/framework/testserver.go:498]: Extracted a helper method that any module can now use 👍 

9080129

File: e2e/framework/testserver.go (hunk 3 above)
Comment (zivkovicmilos) [reply e2e/framework/testserver.go]: Extracted a helper method that any module can now use 👍 

9080129

**Review by lazartravica — APPROVED** [review]

**Pull request merged**